chapter_5/exercise_5_07/readlines.c

```
@@ -16,115 +16,101 @@ void writelines(char *line_ptr[], int nr_of_lines);
-    else
-    {
-      line[len - 1] = '\0'; // Delete newline
-      strcpy(p, line);
-      line_ptr[nr_of_lines++] = p;
-      p += len; // Move p to the next empty position
+int readlines(char *line_ptr[], int max_nr_of_lines, char *stored_lines) {
+    int len;
+    int nr_of_lines;
+
+    char *p =
+        stored_lines + strlen(stored_lines); // Init p with the first empty
```

Cppcheck (2.18.0)

[warning] Uninitialized variable

(uninitvar)